netci.groovy

```diff
@@ -49,71 +49,98 @@ def getDockerImageForOs(os) {
   def imageMap = [
     'RHEL7.2': 'microsoft/dotnet-buildtools-prereqs:rhel7_prereqs_2',
     'CentOS7.1': 'microsoft/dotnet-buildtools-prereqs:centos-7-b46d863-20180719033416',
+    'Fedora28': 'microsoft/dotnet-buildtools-prereqs:fedora-28-c103199-20180628122443',
   ]
   return imageMap.get(os)
 }
 
-def addBuildStepsAndSetMachineAffinity(def job, String os, String configuration) {
+def addBuildStepsAndSetMachineAffinity(def job, String os, String configuration, boolean runInDocker) {
+  smokeTestExcludes = "";
+  if (os == "OSX10.12") {
+    // Dev certs doesn't seem to work in these platforms. https://github.com/dotnet/source-build/issues/560
+    smokeTestExcludes += " --excludeWebHttpsTests";
+  }
+
   job.with {
     steps {
-      if (os == "Windows_NT") {
-        batchFile("git submodule update --init --recursive");
-        batchFile(".\\build.cmd /p:Configuration=${configuration} ${loggingOptions}")
+      if (runInDocker) {
+        def imageName = getDockerImageForOs(os);
+      
+        shell("git submodule update --init --recursive");
+        shell("docker run -u=\"\$(id -u):\$(id -g)\" -t --sig-proxy=true -e HOME=/opt/code/home -v \$(pwd):/opt/code --rm -w /opt/code ${imageName} /opt/code/build.sh /p:Configuration=${configuration} ${loggingOptions}");
+        shell("docker run -u=\"\$(id -u):\$(id -g)\" -t --sig-proxy=true -e HOME=/opt/code/home -v \$(pwd):/opt/code --rm -w /opt/code ${imageName} /opt/code/smoke-test.sh --minimal --configuration ${configuration} ${smokeTestExcludes}");
```

Comment:
These long lines and how much the beginning part is repeated throughout the file bugs me. It seems like there's a lot of opportunity to clean up this script, such as pulling the tarball/offline-tarball steps into functions like this one and sharing some stuff.

On the other hand, time spent cleaning up might be better spent moving to the VSTS instance for public CI. Realized we don't have an issue for that, so filed https://github.com/dotnet/source-build/issues/757.

This doesn't look unmaintainable to me yet, so I'm ok merging as is.